src/migrations/create.rs

```diff
@@ -106,12 +107,11 @@ pub enum MigrationKey {
     Fixup { target_revision: String },
 }
 
-pub trait MigrationToText {
-    type StatementsIter<'a>: Iterator<Item = &'a String> where Self: 'a;
+pub trait MigrationToText<'a, T: Iterator<Item = &'a String> = std::iter::Once<&'a String>> {
     fn key(&self) -> &MigrationKey;
     fn parent(&self) -> anyhow::Result<&str>;
     fn id(&self) -> anyhow::Result<&str>;
-    fn statements<'a>(&'a self) -> Self::StatementsIter<'a>;
+    fn statements(&'a self) -> T;
```

Comment:
For context of this change, see https://github.com/rust-lang/rust/issues/102211#issuecomment-1513931928 & https://edgedb.slack.com/archives/CNECHA9EW/p1712953250921689

Reply:
_a lot of debugging_ ended up the codepath in branch commands that used that trait, and some cheeki googling